src/zui/ZUIAutocompleteInPlaceTest/index.tsx

```diff
@@ -0,0 +1,282 @@
+import Fuse from 'fuse.js';
+import { lighten } from '@mui/material/styles';
+import makeStyles from '@mui/styles/makeStyles';
+import { useState } from 'react';
+import {
+  Autocomplete,
+  Box,
+  ClickAwayListener,
+  IconButton,
+  TextField,
+  Tooltip,
+} from '@mui/material';
+
+import EventTypesModel from 'features/events/models/EventTypesModel';
+import messageIds from 'features/events/l10n/messageIds';
+import useEditPreviewBlock from 'zui/hooks/useEditPreviewBlock';
+import { useMessages } from 'core/i18n';
+import { ZetkinActivity } from 'utils/types/zetkin';
+import ZUIPreviewableInput from 'zui/ZUIPreviewableInput';
+
+const useStyles = makeStyles((theme) => ({
+  create: {
+    display: 'contents',
+    fontSize: '1em',
+    margin: 0,
+    paddingLeft: 0,
+  },
+  inputRoot: {
+    '& fieldset': { border: 'none' },
+    border: '2px dotted transparent',
+    borderColor: lighten(theme.palette.primary.main, 0.65),
+    borderRadius: 10,
+    maxWidth: '200px',
+    overflow: 'hidden',
+    paddingLeft: 10,
+    paddingRight: 0,
+    textOverflow: 'ellipsis',
+    width: '18vw',
+  },
+  preview: {
+    // Same styles as input
+    '& fieldset': { border: 'none' },
+    '&:focus, &:hover': {
+      borderColor: lighten(theme.palette.primary.main, 0.65),
+      paddingLeft: 10,
+      paddingRight: 0,
+    },
+    border: '2px dotted transparent',
+    borderRadius: 10,
+    maxWidth: '200px',
+    transition: 'all 0.2s ease',
+    width: '18vw',
+  },
+  showBorder: {
+    '& fieldset': { border: 'none' },
+    border: '2px dotted transparent',
+    borderColor: lighten(theme.palette.primary.main, 0.65),
+    borderRadius: 10,
+    maxWidth: '200px',
+    paddingLeft: 10,
+    paddingRight: 0,
+    transition: 'all 0.2s ease',
+    width: '18vw',
+  },
+}));
+
+interface NewEventType {
+  id?: number;
+  title: string | null;
+  info_text?: string | null;
+  createType?: boolean;
+}
+
+interface ZUIAutocompleteInPlaceTestProps {
+  currentType: NewEventType;
+  onTypeChange: (value: string) => void;
+  types: ZetkinActivity[];
+  typesModel: EventTypesModel;
+  showBorder: boolean;
+  setAutoOnEdit: (value: boolean) => void;
+  setShowBorder: (value: boolean) => void;
+}
+
+const ZUIAutocompleteInPlaceTest = ({
+  types,
+  currentType,
+  onTypeChange,
+  typesModel,
+  setAutoOnEdit,
+  setShowBorder,
+  showBorder,
+}: ZUIAutocompleteInPlaceTestProps) => {
+  const [editing, setEditing] = useState<boolean>(false);
+  const [eventType, setEventType] = useState<NewEventType>(currentType);
+  const messages = useMessages(messageIds);
+
+  if (currentType === null) {
+    types = [
+      { id: 0, info_text: '', title: messages.type.uncategorized() },
+      ...types,
+    ];
+  }
+  const eventTypes: NewEventType[] = types;
+
+  const classes = useStyles();
+  const fuse = new Fuse(eventTypes, { keys: ['title'], threshold: 0.4 });
+
+  const { clickAwayProps, previewableProps } = useEditPreviewBlock({
```

Comment:
I don't think it should be necessary to use `useEditPreviewBlock`, or even `ZUIPreviewableInput` or `ClickAwayListener`. Note how `ZUIEditTextInPlace` don't use either of those, and still is able to do more or less what this component is trying to do.

In my head, using a regular `<Autocomplete>`, styling it differently in different states, and moving between the states `onFocus` and `onBlur` should be enough. But I haven't tried, so maybe it's not possible. I'm thinking we should take a look at it together!

Reply:
Okay!